packages/@ember/-internals/routing/lib/system/router.ts

```diff
@@ -472,7 +476,9 @@ class EmberRouter extends EmberObject {
       this._toplevelView = OutletView.create();
       this._toplevelView.setOutletState(liveRoutes as GlimmerOutletState);
       let instance: any = owner.lookup('-application-instance:main');
-      instance.didCreateRootView(this._toplevelView);
+      if (instance) {
+        instance.didCreateRootView(this._toplevelView);
+      }
```

Comment:
On a cursory review, this seems _somewhat_ unrelated to the other changes. What circumstances would have forced `_setOutlets` to be called without an `-application-instance:main` (IOW why is this change needed)?

Reply:
hmmm `RenderingTestCase` does not have `-application-instance:main` registered. I guess it's just how ember internal test setup, not having any real impact on user's test. Should we use another base test case class?

Reply:
No, I think its fine but I'd like to make an explicit test case so we don't accidentally make rendering test case have one and remove the guard. Specifically, I think tests that use a custom resolver (e.g. ember-engines's `engineResolverFor`) will have this exact issue.

Reply:
Created a RouterNonApplicationTestCase